packages/odyssey-react/src/components/NativeSelect/NativeSelect.tsx

```diff
@@ -0,0 +1,158 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { useCallback } from "react";
+import { NativeSelectOption } from "./NativeSelectOption";
+import { NativeSelectOptionGroup } from "./NativeSelectOptionGroup";
+import { Field } from "../Field";
+import { ChevronDownIcon } from "../Icon";
+import { withTheme } from "@okta/odyssey-react-theme";
+import { forwardRefWithStatics, useOmit, useOid } from "../../utils";
+import { theme } from "./NativeSelect.theme";
+import styles from "./NativeSelect.module.scss";
+
+import type {
+  ComponentPropsWithRef,
+  ReactNode,
+  ChangeEvent,
+  FocusEventHandler,
+} from "react";
+import type { CommonFieldProps } from "../Field/types";
+
+export interface NativeSelectProps
+  extends CommonFieldProps,
+    Omit<ComponentPropsWithRef<"select">, "style" | "className" | "multiple"> {
+  /**
+   * Select options
+   */
+  children: ReactNode;
+
+  /**
+   * The underlying select element id attribute. Automatically generated if not provided
+   */
+  id?: string;
+  /**
+   * The underlying select element name attribute for the group
+   */
+  name: string;
+
+  /**
+   * The underlying select element disabled attribute for the group
+   * @default false
+   */
+  disabled?: boolean;
+
+  /**
+   * The selected option value attribute for a controlled group.
+   */
+  value?: string;
+
+  /**
+   * Callback executed when the select fires a blur event
+   * @param {Object} event the event object
+   */
+  onBlur?: FocusEventHandler<HTMLSelectElement>;
```

Comment:
is this necessary? I believe we can re-use the prop defs for this callback from the intrinsic `ComponentPropsWithRef<"select">` definition

Reply:
It's not strictly necessary, but without it the callbacks don't surface in the docs. In my previous experience this is helpful and removes ambiguity for consumers. It's consistent with the way TextInput  is implemented https://github.com/okta/odyssey/blob/develop/packages/odyssey-react/src/components/TextInput/TextInput.tsx#L80